### ISSUE TITLE
No async tests with 2 or more computational components

### ISSUE THREAD

**Issue by dqwu** [NCAR/ParallelIO#403]
There is an existing test file test_async_2comp.c, but it is not added for ctest yet. It also has build errors.

For test_async_4proc.c, I tried to modify it to use 2 computational components, but the modified test would fail.
```
/* Number of computational components to create. */
#define COMPONENT_COUNT 2

/* Num procs for IO and computation. */
int num_procs[NUM_COMBOS][COMPONENT_COUNT + 1] = {{2, 1, 1}, {1, 2, 1}, {1, 1, 2}};

/* Number of processors that will do IO. */
int num_io_procs[NUM_COMBOS] = {2, 1, 1};
```

**Comment by edhartnett** [comment]
Yes I quite agree. I will get test_async_2comp.c working as a test. Actually I was not aware that I had added that to the repo. It shouldn't be there until it is a working test. But for now I will just get it working.

Thanks!

**Comment by edhartnett** [comment]
Work on this will continue after I get #1134 done.
